File: springboot/spring data jpa for sqlite/sqlite/src/main/java/com/weilai/sqlite/controller/UserController.java
package com.weilai.sqlite.controller;

import com.weilai.sqlite.entity.User;
import com.weilai.sqlite.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class UserController {
    @Autowired
    private UserRepository userRepository;


    @GetMapping("/users")
    public List<User> getUsers() {
        return userRepository.findAll();
    }

    @PostMapping("/save-user")
    public User saveUser(@RequestBody User user) {
        return userRepository.save(user);
    }
}

File: kotlin/spring data jpa for kotlin sqlite/sqliteKt/src/main/kotlin/com/weilai/sqliteKt/controller/UserController.kt
package com.weilai.sqliteKt.controller

import com.weilai.sqliteKt.entity.User
import com.weilai.sqliteKt.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
class UserController(
    @Autowired private val userRepository: UserRepository
) {

    @GetMapping("/users")
    fun getUsers(): MutableList<User> = userRepository.findAll()

    @PostMapping("/save-user")
    fun saveUser(@RequestBody user: User): User {
        user.createdAt = LocalDateTime.now()
        return userRepository.save(user)
    }
}